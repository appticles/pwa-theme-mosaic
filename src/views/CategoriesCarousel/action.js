import config from 'react-global-configuration';

export const REQUEST_CATEGORIES = 'REQUEST_CATEGORIES';
export const RECEIVE_CATEGORIES = 'RECEIVE_CATEGORIES';

export const requestCategories = () => ({
  type: REQUEST_CATEGORIES,
});

export const receiveCategories = categories => ({
  type: RECEIVE_CATEGORIES,
  categories,
});

export const fetchCategories = (params = {}) => (dispatch) => {
  dispatch(requestCategories());

  let url;
  if (params && params.post) {
    url = config.get('export').categories + '?post=' + String(params.post);
  } else {
    url =
      config.get('export').categories +
      '?' +
      Object.keys(params)
        .map(k => k + '=' + encodeURIComponent(params[k]))
        .join('&');
  }
  return fetch(url)
    .then(response => response.json())
    .then(json => dispatch(receiveCategories(json)))
    .catch(() => {
      dispatch(receiveCategories([]));
    });
};
