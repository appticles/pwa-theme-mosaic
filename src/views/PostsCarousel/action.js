import config from 'react-global-configuration';

export const REQUEST_POSTS = 'REQUEST_POSTS';
export const RECEIVE_POSTS = 'RECEIVE_POSTS';

export const requestPosts = () => ({
  type: REQUEST_POSTS,
});

export const receivePosts = posts => ({
  type: RECEIVE_POSTS,
  posts,
});

export const fetchPosts = (params = {}) => (dispatch) => {
  dispatch(requestPosts());
  config.set(window.__INITIAL_CONFIG__, { freeze: false });

  let url;
  if (params && params.id) {
    url = config.get('export').posts + '/' + String(params.id);
  } else {
    url =
      config.get('export').posts +
      '?' +
      Object.keys(params)
        .map(k => k + '=' + encodeURIComponent(params[k]))
        .join('&');
  }
  return fetch(url)
    .then(response => response.json())
    .then(json => dispatch(receivePosts(json)))
    .catch(() => {
      dispatch(receivePosts([]));
    });
};
