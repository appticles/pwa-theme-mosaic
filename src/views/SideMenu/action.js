import config from 'react-global-configuration';

export const REQUEST_PAGES = 'REQUEST_PAGES';
export const RECEIVE_PAGES = 'RECEIVE_PAGES';

export const requestPages = () => ({
  type: REQUEST_PAGES,
});

export const receivePages = pages => ({
  type: RECEIVE_PAGES,
  pages,
});

export const fetchPages = (params = {}) => (dispatch) => {
  dispatch(requestPages());

  let url;
  if (params && params.id) {
    url = config.get('export').pages + '/' + String(params.id);
  } else {
    url =
      config.get('export').pages +
      '?' +
      Object.keys(params)
        .map(k => k + '=' + encodeURIComponent(params[k]))
        .join('&');
  }
  return fetch(url)
    .then(response => response.json())
    .then(json => dispatch(receivePages(json)))
    .catch(() => {
      dispatch(receivePages([]));
    });
};
