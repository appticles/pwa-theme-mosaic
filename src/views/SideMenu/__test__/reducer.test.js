import pages, { INITIAL_STATE, getPages, getPagesFetching } from '../reducer';
import { requestPages, receivePages } from '../action';

describe('Posts reducer', () => {
  it('returns same state on an unhandled action', () => {
    expect(pages(INITIAL_STATE, { type: '_NULL' })).toMatchSnapshot();
  });
  it('handles REQUEST_PAGES action', () => {
    const pageState = {
      isFetching: 0,
      items: [
        {
          id: 1,
          name: 'Nice page',
        },
        {
          id: 2,
          name: 'Nicer page',
        },
      ],
    };
    expect(pages(pageState, requestPages())).toMatchSnapshot();
  });
  it('handles RECEIVE_PAGES action case 1', () => {
    const pageState = {
      isFetching: 1,
      items: [
        {
          id: 1,
          name: 'Nice page',
        },
        {
          id: 2,
          name: 'Nicer page',
        },
      ],
    };
    const pageList = {
      id: 3,
      name: 'Nicest page',
    };
    expect(pages(pageState, receivePages(pageList))).toMatchSnapshot();
  });
  it('handles RECEIVE_PAGES action case 2', () => {
    const pageState = {
      isFetching: 1,
      items: [
        {
          id: 1,
          name: 'Nice page',
        },
        {
          id: 2,
          name: 'Nicer page',
        },
      ],
    };
    const pageList = [
      {
        id: 3,
        name: 'Nicest page',
      },
    ];
    expect(pages(pageState, receivePages(pageList))).toMatchSnapshot();
  });

  // reducer functions
  it('getPagesFetching returns state.isFetching', () => {
    expect(getPagesFetching(INITIAL_STATE)).toBe(0);
  });
  it('getPages returns state.items', () => {
    const state = {
      items: [
        {
          name: 'Some page',
        },
      ],
    };
    expect(getPages(state)).toBe(state.items);
  });
});
