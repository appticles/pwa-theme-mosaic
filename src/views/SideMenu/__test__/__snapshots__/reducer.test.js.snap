// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts reducer handles RECEIVE_PAGES action case 1 1`] = `
Object {
  "isFetching": 0,
  "items": Array [
    Object {
      "id": 1,
      "name": "Nice page",
    },
    Object {
      "id": 2,
      "name": "Nicer page",
    },
    Object {
      "id": 3,
      "name": "Nicest page",
    },
  ],
}
`;

exports[`Posts reducer handles RECEIVE_PAGES action case 2 1`] = `
Object {
  "isFetching": 0,
  "items": Array [
    Object {
      "id": 1,
      "name": "Nice page",
    },
    Object {
      "id": 2,
      "name": "Nicer page",
    },
    Object {
      "id": 3,
      "name": "Nicest page",
    },
  ],
}
`;

exports[`Posts reducer handles REQUEST_PAGES action 1`] = `
Object {
  "isFetching": 1,
  "items": Array [
    Object {
      "id": 1,
      "name": "Nice page",
    },
    Object {
      "id": 2,
      "name": "Nicer page",
    },
  ],
}
`;

exports[`Posts reducer returns same state on an unhandled action 1`] = `
Object {
  "isFetching": 0,
  "items": Array [],
}
`;
